src/internal/api/packages/find.go

```diff
@@ -44,11 +44,44 @@ func findPackage(pattern *regexp.Regexp, w http.ResponseWriter, setDir func() (s
 		return
 	}
 
-	files, err := utils.RecursiveFileList(targetDir, pattern)
+	files, err := recursiveFileListSkipPermissionErrors(targetDir, pattern)
 	if err != nil || len(files) == 0 {
 		pkgNotFoundMsg := fmt.Sprintf("Unable to locate the package: %s", pattern.String())
 		message.ErrorWebf(err, w, pkgNotFoundMsg)
 		return
 	}
 	common.WriteJSONResponse(w, files, http.StatusOK)
 }
+
+// RecursiveFileList walks a path with an optional regex pattern and returns a slice of file paths.
+func recursiveFileListSkipPermissionErrors(dir string, pattern *regexp.Regexp) (files []string, err error) {
+	const dotcharater = 46
+	err = filepath.WalkDir(dir, func(path string, d fs.DirEntry, err error) error {
+		// Return errors
+		if err != nil {
+			// skip files and dirs we don't have permission to view.
+			if os.IsPermission(err) {
+				return nil
+			}
+			return err
+		}
+
+		// Skip hidden directories
+		if d.IsDir() && d.Name()[0] == dotcharater {
```

Comment:
This will skip the default `~/.zarf-cache`, which is where the current init package is cached (and where future caching will be defaultly located)

Reply:
Just curious when it begins being stored in the .zarf-cache is that a recent change? as this is derived from the io.go ListDirectorys that was previously here and didn't work but it currently is not having an issue. Was just curious about the inconsistency there and wether or not that ListDirectories method needs to be updated as well if the .zarf-cache is going to need to be explorable.

Reply:
Or is the .zarf-cache used when running through the brew package and not compiled from source?

Reply:
No the `~/.zarf-cache` is the default location no matter where Zarf is built from and is governed by the `--zarf-cache` CLI flag / configuration.  When you do a `zarf init` and choose to download the init package, it will store it inside this folder. Currently, zarf packages are not stored in the cache as they are spit out to the location specified w/ the `-o` flag on `zarf package create`.

Reply:
This list packages function will also be extended in the future to query OCI registries for Zarf packages stored.

Reply:
I think there might be bigger issues that will have to be looked into because the zarf cache gives this error
Unable to read the package at: %2FUsers%2Fmikewinberry%2F.zarf-cache%2Fzarf-init-amd64-v0.25.0-20-g75b142a4.tar.zst Its the only package that seems to be unreadable.

Reply:
Disregard seems to have been an issue with an old zarf-init package in my cache.